pageserver/src/tenant/layer_map.rs

```diff
@@ -28,11 +27,12 @@ use std::sync::Arc;
 use tracing::*;
 use utils::lsn::Lsn;
 
+use super::storage_layer::PureLayer;
+
 ///
 /// LayerMap tracks what layers exist on a timeline.
 ///
-#[derive(Default)]
-pub struct LayerMap {
+pub struct LayerMap<PersistentLayerT: ?Sized> {
```

Comment:
NIT: We dont have any sort of convention for naming of generic parameters, I can see what is the reason behind trailing T, I would still rather leave the parameter name without suffixes/prefixes. But that only my personal preference, feel free to ignore

Reply:
I also disagree with the naming, given that `PersistentLayerT: ?Sized + PureLayer` below and, in our terms, the persistent part is `Layer` since only that has methods like `filename`.
I've renamed `PersistentLayerT` to just `L` and will rename the traits: `PureLayer` -> `Layer`, `Layer` -> `PersistentLayer` in the next commit.